tsz-compress/src/v2/mod.rs

```diff
@@ -1,18 +1,1355 @@
+
+        let mut iter = self.iter();
+        while let Some(value) = iter.next() {
+            let remaining = iter.size_hint().0;
+            let index = queue_length - remaining;
+
+            if ((0..=2).contains(&index) && !(-32768..=32767).contains(&value)) {
+                sixteen = false;
+            }
+            if ((0..=3).contains(&index) && !(-512..=511).contains(&value)) {
+                ten = false;
+            }
+            if ((0..=4).contains(&index) && !(-128..=127).contains(&value)) {
+                eight = false;
+            }
+            if ((0..=8).contains(&index) && !(-8..=7).contains(&value)) {
+                four = false;
+            }
+            if ((0..=10).contains(&index) && !(-4..=3).contains(&value)) {
+                three = false;
+            }
+        }
+
+        // Emit according to priority of cases
+        if three {
+            push_header_pad_three_bits(out);
+            for _ in 0..10 {
+                if let Some(value) = self.pop() {
+                    let value = (value << 1i64) ^ (value >> 63i64); // ZigZag Encoding
+                    value.extend_bits(0..3, out);
+                }
+            }
+            return 10;
+        } else if four {
+            push_header_pad_four_bits(out);
+            for _ in 0..8 {
+                if let Some(value) = self.pop() {
+                    let value = (value << 1i64) ^ (value >> 63i64); // ZigZag Encoding
+                    value.extend_bits(0..4, out);
+                }
+            }
+            return 8;
+        } else if eight {
+            push_header_pad_eight_bits(out);
+            for _ in 0..4 {
+                if let Some(value) = self.pop() {
+                    let value = (value << 1i64) ^ (value >> 63i64); // ZigZag Encoding
+                    value.extend_bits(0..8, out);
+                }
+            }
+            return 4;
+        } else if ten {
+            push_header_pad_ten_bits(out);
+            for _ in 0..3 {
+                if let Some(value) = self.pop() {
+                    let value = (value << 1i64) ^ (value >> 63i64); // ZigZag Encoding
+                    value.extend_bits(0..10, out);
+                }
+            }
+            return 3;
+        } else if sixteen {
+            push_header_pad_sixteen_bits(out);
+            for _ in 0..2 {
+                if let Some(value) = self.pop() {
+                    let value = (value << 1i64) ^ (value >> 63i64); // ZigZag Encoding
+                    value.extend_bits(0..16, out);
+                }
+            }
+            return 2;
+        }
+        0
+    }
+}
+
+impl EmitDeltaBits<i32> for CompressionQueue<i32, 10> {
+    #[allow(unused)]
+    fn emit_delta_bits(&mut self, out: &mut BitBuffer, flush: bool) -> usize {
+        let mut iter = self.iter();
+        let mut three = true;
+        let mut four = true;
+        let mut eight = true;
+        let mut ten = true;
+        let mut sixteen = true;
```

Comment:
If the queue is full or flushing and the delta bits cannot fit the data, we will need to emit the delta-delta bits in order to still represent the data. This can only happen on the i32 and i64 case.